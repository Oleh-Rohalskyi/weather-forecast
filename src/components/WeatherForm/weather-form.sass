@import values

.weather-form
    &__label
        display: block
        text-align: center
        font-family: 'Nunito'
        font-size: 16px
        letter-spacing: 0
        color: #114358
        &.danger 
            color: #CE4214
    &__input
        width: 325px
        height: 45px
        border-radius: 2px
        border: 2px solid #17475B
        display: block
        margin: 14px auto 26px auto
        text-align: center
        font-family: 'Roboto'
        font-size: 16px
    &__submit
        margin: 0 auto
        display: block
        width: 160px
        height: 40px
        background-color: #F3997B
        box-shadow: 0px 3px 6px rgba(0, 0, 0, 0.1)
        border-radius: 3px
        text-align: center
        font-family: 'Roboto'
        letter-spacing: 0
        font-weight: bold
        margin-bottom: 50px
        color: rgba(255, 255, 255, 1)
    @media (max-width: $min-size)
        &__input
            width: 234px
            margin: 14px auto 16px auto