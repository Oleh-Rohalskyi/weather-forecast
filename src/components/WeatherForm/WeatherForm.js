import React, { useState, useEffect } from 'react';
import {useDispatch} from 'react-redux';
import {actionTypes} from '../../features/weather';

import './weather-form.sass';

const WeatherFrom = ({name}) => {
  
  const [city, setCity] = useState("");
  
  const dispatch = useDispatch();

  const handleClick = (e) => {
      e.preventDefault();
      dispatch(actionTypes.weatherCharterer(actionTypes.CALL_FROM_WEATHER_FORM))
      dispatch(actionTypes.fetchWeather(city))
  }

  const handleInputChange = (e) => {
    setCity(e.target.value)
  }

  return (
      <form className="weather-form">
        <label className="weather-form__label"  htmlFor={name}>Please enter a city</label>
        <input className="weather-form__input" type="text" id={name} onChange={handleInputChange}/>
        <button className="weather-form__submit" onClick={handleClick}>Have a look</button>
      </form>
  );
};

export default WeatherFrom;
