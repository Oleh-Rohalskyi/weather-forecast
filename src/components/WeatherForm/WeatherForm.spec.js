import { render, fireEvent, getByTestId, act} from "@testing-library/react-hooks";

import React from 'react';
import { Provider } from 'react-redux';
import { mount } from 'enzyme';
import configureStore from 'redux-mock-store';
import WeatherForm from './WeatherForm';


describe('src > components > WeatherForm', () => {

  const mockUp = {}


  const mockStore = configureStore(mockUp);
  const store = mockStore(mockUp);

  jest.spyOn(store, 'dispatch');

  beforeEach(() => {
    store.dispatch.mockClear();
  });

  it('renders without crashing', () => {
    
    const wrapper = mount(
      <Provider store={store}>
        <WeatherForm name="test-form" />
      </Provider>
    );
    // wrapper.find('button').simulate("click")
    // wrapper.debug();
    // console.log(wrapper)
    expect(wrapper).toMatchSnapshot();

  });

});
