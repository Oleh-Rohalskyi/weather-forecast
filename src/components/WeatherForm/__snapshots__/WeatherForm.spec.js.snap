// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`src > components > WeatherForm renders without crashing 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [MockFunction],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <WeatherFrom
    name="test-form"
  >
    <form
      className="weather-form"
    >
      <label
        className="weather-form__label"
        htmlFor="test-form"
      >
        Please enter a city
      </label>
      <input
        className="weather-form__input"
        id="test-form"
        onChange={[Function]}
        type="text"
      />
      <button
        className="weather-form__submit"
        onClick={[Function]}
      >
        Have a look
      </button>
    </form>
  </WeatherFrom>
</Provider>
`;
